godtools/App/Services/Renderer/Views/MobileContent/Views/Pages/MobileContentPagesViewModel.swift

```diff
@@ -377,8 +377,19 @@ class MobileContentPagesViewModel: NSObject, ObservableObject {
         switch initialPage {
         
         case .pageId(let value):
-            return allPages.first(where: {$0.id == value})
```

Comment:
I need to think about this logic a little bit.  I'm not 100% certain if we want this change here in the base class.

Reply:
Was this added because we don't want to track a hidden page as part of lesson progress?

Reply:
We can track a hidden page, we just don't want to open a hidden page in the case that a user switches languages and reopens a lesson, but the page they were on in the old language is hidden in the new language.  If that happens, we instead show the first non-hidden previous page.

Also if the user stops on the last visible page, we automatically restart at the first page when they reopen the lesson.